security/pkg/pki/ca/ca_test.go

```diff
@@ -158,7 +157,7 @@ func TestCreateSelfSignedIstioCAWithSecret(t *testing.T) {
 	signingKeyPem := []byte(key1Pem)
 
 	client := fake.NewSimpleClientset()
-	initSecret := k8ssecret.BuildSecret("", CASecret, "default",
+	initSecret := BuildSecret("", CASecret, "default",
```

Comment:
The old code seemed right to me. Kubernetes specific abstractions (Secret) are decoupled from core code.

Reply:
Not touching K8s Secret abstraction any more and make it self-contained. Would you please take another look @howardjohn? Thanks!